### PR TITLE
fix: Api Refactor Create/Revoke and Filter Resource AM

### PR DESCRIPTION
**Pull request by abhijitdiwan**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/AM-110

### Change description ###
fix: Api Refactor Create/Revoke and Filter Resource AM

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/amapi/controllers/AccessManagementController.java [hunk 1]
```diff
@@ -70,7 +70,7 @@
                                      @RequestHeader(name = "Caller", defaultValue = "Anonymous") String caller) {
         try {
             MDC.put("caller", caller);
-            am.revokeResourceAccess(request);
+            accessManagementService.revokeResourceAccess(request);
             return new ResponseEntity<>(NO_CONTENT);
         } finally {
             MDC.clear();
```
Comment (shashank-rastogi) [inline src/main/java/uk/gov/hmcts/reform/amapi/controllers/AccessManagementController.java]: do we really need MDC to set the caller on the apis? can you investigate this?

File: src/main/java/uk/gov/hmcts/reform/amapi/controllers/AccessManagementController.java [hunk 2]
```diff
@@ -48,7 +48,7 @@
                                      @RequestHeader(name = "Caller", defaultValue = "Anonymous") String caller) {
```
Comment (shashank-rastogi) [inline src/main/java/uk/gov/hmcts/reform/amapi/controllers/AccessManagementController.java]: API end points needs to be changed as they are documented in confluence

File: src/test/java/uk/gov/hmcts/reform/amapi/controllers/AccessManagementControllerTest.java [hunk 3]
```diff
@@ -100,7 +100,7 @@ public void testRevokeResourceAccess() throws Exception {
 
         doNothing().when(accessManagementService).revokeResourceAccess(any());
 
-        this.mockMvc.perform(delete("/lib/revoke-resource-access")
+        this.mockMvc.perform(delete("/api/revoke-resource-access")
```
Comment (shashank-rastogi) [inline src/test/java/uk/gov/hmcts/reform/amapi/controllers/AccessManagementControllerTest.java]: Change the rest api end point to Delete /access-resource

File: src/main/java/uk/gov/hmcts/reform/amapi/controllers/AccessManagementController.java (hunk 1 above)
Comment (abhijitdiwan) [reply src/main/java/uk/gov/hmcts/reform/amapi/controllers/AccessManagementController.java]: Removed

File: src/main/java/uk/gov/hmcts/reform/amapi/controllers/AccessManagementController.java (hunk 2 above)
Comment (abhijitdiwan) [reply src/main/java/uk/gov/hmcts/reform/amapi/controllers/AccessManagementController.java]: done

File: src/test/java/uk/gov/hmcts/reform/amapi/controllers/AccessManagementControllerTest.java (hunk 3 above)
Comment (abhijitdiwan) [reply src/test/java/uk/gov/hmcts/reform/amapi/controllers/AccessManagementControllerTest.java]: done

**Comment by abhijitdiwan** [comment]
Api Refactor Create/Revoke and Filter Resource AM-110

**Pull request merged**